internal/orchestrator/orchestrator.go

```diff
@@ -359,14 +364,24 @@ func (or *orchestrator) initPlugins(ctx context.Context) (err error) {
 		return err
 	}
 
-	if or.publicstorage == nil {
-		psType := config.GetString(config.PublicStorageType)
-		if or.publicstorage, err = psfactory.GetPlugin(ctx, psType); err != nil {
-			return err
+	if or.sharedstorage == nil {
+		ssType := config.GetString(config.SharedStorageType)
```

Comment:
You might have a look at the `initDataExchange` method, which does a similar backwards-compatible plugin initialization. In particular, I'd suggest simply checking if `ssType` is set or empty. If it's set, it needs to be valid. If it's empty, fallback to check the old config name. I don't think it's worth calling `GetPlugin` with an empty string and then looking for errors.